public/coffee/ide/pdf/controllers/PdfController.coffee

```diff
@@ -73,6 +74,27 @@ define [
 			$scope.pdf.view = 'errors'
 			$scope.pdf.renderingError = true
 
+		triggerAutoCompile = () ->
+			if (!ide.$scope.hasLintingError)
+				$scope.recompile()
+
+		autoCompileListener = null
+		toggleAutoCompile = (enabling) ->
```

Comment:
I'm not sure if I made this overly complicated but I wanted to try tearing down the listener if auto-compile is disabled. This prevents unnecessary calls, but might be pre-optimisation.

Happy to remove if you disagree

Reply:
Yup it's good to clean things up.